Source/WebKit/UIProcess/Cocoa/MediaPermissionUtilities.mm

```diff
@@ -190,6 +190,18 @@ bool checkUsageDescriptionStringForSpeechRecognition()
 
 void alertForPermission(WebPageProxy& page, MediaPermissionReason reason, const WebCore::SecurityOriginData& origin, CompletionHandler<void(bool)>&& completionHandler)
 {
+    if (!isMainRunLoop()) {
+        callOnMainRunLoop([weakPage = WeakPtr { page }, reason, origin, completionHandler = WTFMove(completionHandler)]() mutable {
```

Comment:
origin would need to be isolated copied.

But this is not the only issue, we just shouldn't be using a WebPageProxy from a non main thread, the `WeakPtr { page }` wouldn't be safe either.